src/rdkafka.h

```diff
@@ -2043,11 +2043,54 @@ rd_kafka_t *rd_kafka_new(rd_kafka_type_t type, rd_kafka_conf_t *conf,
  * @brief Destroy Kafka handle.
  *
  * @remark This is a blocking operation.
+ * @remark rd_kafka_consumer_close() will be called from this function
+ *         if the instance type is RD_KAFKA_CONSUMER, a \c group.id was
+ *         configured, and the rd_kafka_consumer_close() was not
+ *         explicitly called by the application. This in turn may
+ *         trigger consumer callbacks, such as rebalance_cb.
+ *         Use rd_kafka_destroy_flags() with
+ *         RD_KAFKA_DESTROY_F_NO_CONSUMER_CLOSE to avoid this behaviour.
+ *
+ * @sa rd_kafka_destroy_flags()
  */
 RD_EXPORT
 void        rd_kafka_destroy(rd_kafka_t *rk);
 
 
+/**
+ * @brief Destroy Kafka handle according to specified destroy flags
+ *
+ */
+RD_EXPORT
+void rd_kafka_destroy_flags (rd_kafka_t *rk, int flags);
```

Comment:
i guess this name is ok. the `Ex` extension used by MS has the advantage that most people looking at it will know it's a new improved version of an older method by convention whereas the word flags is specific to this case. it also potentially reads as 'destroying the flags'. On the other hand, i'm not confused.